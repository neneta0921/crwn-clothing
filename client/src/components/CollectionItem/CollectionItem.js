import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cartActions';

import {
  AddButton,
  BackgroundImage,
  CollectionItemContainer,
  CollectionFooterContainer,
  NameContainer,
  PriceContainer,
} from './CollectionItemStyles';

export const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item;
  return (
    <CollectionItemContainer>
      <BackgroundImage key={item.id} className="background-image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer key={item.id} className="name-container">
          {name}
        </NameContainer>
        <PriceContainer key={item.id} className="price-container">
          {price}
        </PriceContainer>
      </CollectionFooterContainer>
      <AddButton key={item.id} className="add-button" onClick={() => addItem(item)} inverted>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
