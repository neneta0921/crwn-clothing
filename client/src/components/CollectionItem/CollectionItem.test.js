import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { CollectionItem } from './CollectionItem';

describe('CollectionItem component', () => {
  let wrapper;
  let mockAddItem;
  const imageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 20;

  beforeEach(() => {
    mockAddItem = jest.fn();
    const mockProps = {
      item: {
        imageUrl: imageUrl,
        price: mockPrice,
        name: mockName,
      },
      addItem: mockAddItem,
    };
    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('should render CollectionItem component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call addItem when AddButton clicked', () => {
    wrapper.find('AddButton').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should render imageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(imageUrl);
  });

  it('should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockName);
  });

  it('should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text());
    expect(price).toBe(mockPrice);
  });
});
