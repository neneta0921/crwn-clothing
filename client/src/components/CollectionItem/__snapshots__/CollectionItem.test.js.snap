// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem component 1`] = `
<div
  className="sc-gsTCUz bLLZpE"
>
  <div
    className="sc-hKgILt hkHzXq"
  />
  <div
    className="sc-eCssSg jFCvIw"
  >
    <span
      className="sc-jSgupP jbqeKv"
    >
      black hat
    </span>
    <span
      className="sc-gKsewC kIfwPX"
    >
      20
    </span>
  </div>
  <button
    className="sc-bdfBwQ hBuaMb sc-dlfnbm kzeJXE"
    onClick={[Function]}
  >
    Add to cart
  </button>
</div>
`;
